node/silkworm/downloader/sentry_client.cpp

```diff
@@ -45,21 +47,28 @@ void SentryClient::subscribe(Scope scope, subscriber_t callback) { subscribers_[
 
 void SentryClient::publish(const sentry::InboundMessage& message) {
     auto subscribers = subscribers_[scope(message)];
-    for (auto& subscriber : subscribers) {
+    for (auto& subscriber: subscribers) {
         subscriber(message);
     }
 }
 
 void SentryClient::set_status(Hash head_hash, BigInt head_td, const ChainIdentity& chain_identity) {
     rpc::SetStatus set_status{chain_identity, head_hash, head_td};
     exec_remotely(set_status);
-
     log::Info() << "SentryClient, set_status sent";
-    sentry::SetStatusReply reply = set_status.reply();
+}
+
+
+void SentryClient::hand_shake() {
+    rpc::HandShake hand_shake;
+    exec_remotely(hand_shake);
+
+    log::Info() << "SentryClient, hand_shake sent";
+    sentry::HandShakeReply reply = hand_shake.reply();
 
     sentry::Protocol supported_protocol = reply.protocol();
     if (supported_protocol != sentry::Protocol::ETH66) {
-        log::Critical() << "SentryClient: sentry do not support eth/66 protocol, is_stopping...";
+        log::Critical() << "SentryClient: sentry do not support eth/66 protocol, is stopping...";
         stop();
         throw SentryClientException("SentryClient exception, cause: sentry do not support eth/66 protocol");
```

Comment:
I find this technique to log::Critical **and** throw exception a bit redundant.
If we throw an exception let the higher level code handle it and print the proper message or don't throw at all and simply log and exit the function

Reply:
Right now the downloader is like an engine that still needs to be installed on the platform. When we will connect the downloader with the stage-sync loop we will address those issues. I think that the stage-sync will dictate to each stage proper error handling and reporting, final result reporting and so on. 

Regarding the cmd utility that I use for test, sure, it is redundant that it catches the error and logs it, al last the cmd utility should report a more high-level error, something like "sentry failed" or "downloader failed".